webtau-http/src/test/groovy/org/testingisdocumenting/webtau/http/datanode/StructuredDataNodeTest.groovy

```diff
@@ -196,4 +196,21 @@ class StructuredDataNodeTest {
         has = node.has('key2.name.foo')
         has.should == false
     }
+
+    @Test
+    void "check level is recorded for non-existent child nodes"() {
+        def node = DataNodeBuilder.fromMap(new DataNodeId("body"), [:])
+        node["key"].should == null
+        node["key"].traceableValue.checkLevel.should == CheckLevel.ExplicitPassed
+
+        node = DataNodeBuilder.fromList(new DataNodeId("body"), [])
+        node["key"].should == null
+        node["key"].traceableValue.checkLevel.should == CheckLevel.ExplicitPassed
+
+        node = DataNodeBuilder.fromValue(new DataNodeId("body"), null)
+        node.should == null
+        node.traceableValue.checkLevel.should == CheckLevel.ExplicitPassed
+        node["key"].should == null
+        node["key"].traceableValue.checkLevel.should == CheckLevel.ExplicitPassed
```

Comment:
very nice.
what about nested maps?

Reply:
I assume you mean this test: node["key"]["nested"].

But do you mean with this node:
```groovy
def node = DataNodeBuilder.fromMap(new DataNodeId("body"), [:])
```

Or this one:
```groovy
def node = DataNodeBuilder.fromMap(new DataNodeId("body"), [key: []])
```

Reply:
OK, this fails:
```groovy
        def node = DataNodeBuilder.fromMap(new DataNodeId("body"), [:])
        node["key"].should == null
        node["key"].traceableValue.checkLevel.should == CheckLevel.ExplicitPassed
        node["key"]["nested"].should == null
        node["key"]["nested"].traceableValue.checkLevel.should == CheckLevel.ExplicitPassed
```

I'll fix.